packages/amplify-app/src/index.js

```diff
@@ -102,42 +102,34 @@ async function amplifyCLIVersionCheck() {
 If not - then generate a skeleton with a base project */
 
 async function createAmplifySkeletonProject() {
+  let frontend;
   if (!fs.existsSync('./amplify')) {
     console.log(`${emoji.get('guitar')} Creating base Amplify project`);
-
-    return new Promise((resolve, reject) => {
-      const amplifyCmd = path.basename(process.argv[1]) === 'amplify-app-dev' ? amplifyDev : amplify;
-      const createSkeletonAmplifyProject = spawn(amplifyCmd, ['init', '--quickstart'], {
-        cwd: process.cwd(),
-        env: process.env,
-        stdio: 'inherit',
-      });
-
-      createSkeletonAmplifyProject.on('exit', code => {
-        if (code === 0) {
-          return getProjectConfig()
-            .then(projectConfig => {
-              const projectConfigFilePath = path.join('amplify', '.config', 'project-config.json');
-              fs.writeFileSync(projectConfigFilePath, JSON.stringify(projectConfig, null, 4));
-              console.log(`${emoji.get('boom')} Successfully created base Amplify Project`);
-              resolve(projectConfig.frontend);
-            })
-            .catch(e => {
-              reject(e);
-            });
-        }
-        console.log(`${emoji.get('x')} Failed to create base Amplify Project`);
-        reject();
-      });
-    });
+    try {
+      await callAmplify(['init', '--quickstart']);
+      const projectConfig = await getProjectConfig();
+      const projectConfigFilePath = path.join('amplify', '.config', 'project-config.json');
+      fs.writeFileSync(projectConfigFilePath, JSON.stringify(projectConfig, null, 4));
+      if (!!projectConfig.javascript) {
+        initializeAwsExports(path.resolve('src'));
+      }
+      console.log(`${emoji.get('boom')} Successfully created base Amplify Project`);
+      frontend = projectConfig.frontend;
+    } catch (err) {
+      console.log(`${emoji.get('x')} Failed to create base Amplify Project`);
+      throw new Error(err);
+    }
+  } else {
+    console.log(
+      `An Amplify project is already initialized in your current working directory ${emoji.get('smiley')}. Not generating base project.`,
+    );
+    console.log();
+    const existingApp = true;
```

Comment:
Nit that comes from the existing code, `existingApp` isn't necessary since it's always `true` and is only used as input in `getProjectConfig(existingApp)`

Reply:
this whole file is being refactored in this PR: https://github.com/aws-amplify/amplify-cli/pull/5977